### PR TITLE
fix(gatsby-transformer-sharp): Add `checkSupportedExtensions` option

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

Initially the idea was to traverse the GraphQL AST and check whether `publicURL` is used alongside `childImageSharp` or not. However, this task would have been really cumbersome and the end result would be brittle as multiple cases (`file`, `allFile`, fragments and more) would need to be handled.
We can revisit this idea once we can return components from GraphQL - I added a `TODO` for that.

In the meantime we expose an option to turn of the warning. The warning is still valuable but if people are sure that they handle all cases, they can turn it off.

### Documentation

The option was documented in the README.

## Related Issues

Follow-up of https://github.com/gatsbyjs/gatsby/pull/20782
Fixes https://github.com/gatsbyjs/gatsby/issues/21823

### CODE REVIEW

File: packages/gatsby-transformer-sharp/README.md [hunk 1]
```diff
@@ -38,6 +38,28 @@ It recognizes files with the following extensions as images.
 
 Each image file is parsed into a node of type `ImageSharp`.
 
+## Configuration options
+
+`checkSupportedExtensions` [boolean][optional]
+
+Sharp only supports certain image formats (see Parsing algorithm) and hence throws a warning when you e.g. use a .gif in an `ImageSharp` query. You'll need to use `publicURL` instead. With this option you can disable the warning behavior.
```
Comment (laurieontech) [inline packages/gatsby-transformer-sharp/README.md]:
```suggestion
Sharp only supports certain image formats (see Parsing algorithm) and hence throws a warning when you e.g. use a .gif in an `ImageSharp` query. You'll need to use `publicURL` instead. With this option you can disable the warning behavior.
```
Where is the parsing algorithm? Can we link to it?

Comment (LekoArts) [reply packages/gatsby-transformer-sharp/README.md]: It's the paragraph above of this section :)

Comment (laurieontech) [reply packages/gatsby-transformer-sharp/README.md]: Haha of course it is. Maybe we can make it (see the Parsing algorithm section above)

**Comment by LekoArts** [comment]
Published in `gatsby-transformer-sharp@2.4.3`

**Pull request merged**